lib/spack/spack/util/executable.py

```diff
@@ -291,6 +291,13 @@ def which_string(*args, **kwargs):
             win_candidates = [name + ext for ext in [".exe", ".bat"]]
         candidate_names = [name] if not win_candidates else win_candidates
 
+        if sys.platform == "win32":
+            new_path = path[:]
+            for p in path:
+                if os.path.basename(p) == "bin":
+                    new_path.append(os.path.dirname(p))
```

Comment:
Doesn't this need an `else` case? If the path doesn't end in `bin`, I don't think we should just drop it.

Reply:
The new_path list always contains the original contents of path, hence the init with `path[:]`, then if it ends in `bin` it gets duplicated without the `bin` on the end.